.github/workflows/gradle.yml

```diff
@@ -292,11 +292,10 @@ jobs:
           aws-access-key-id: ${{ secrets.SELF_RUNNER_AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.SELF_RUNNER_AWS_SECRET_ACCESS_KEY }}
           github-token: ${{ secrets.SELF_RUNNER_GITHUB_ACCESS_TOKEN_1 }}
-  frontend-test:
-    name: "Frontend: Run Test"
-    needs: start-frontend-test-runner # required to start the main job when the runner is ready
-    runs-on: ${{ needs.start-frontend-test-runner.outputs.label }} # run the job on the newly created runner
-    timeout-minutes: 120
+  frontend-prepare:
```

Comment:
nit: I recommend combing the `frontend-prepare` with the `frontend-build` job. I don't think there is value to separating these out as jobs. This helps declutter the github actions summary & simplifies the `needs` block in the stop runner job.